wave_containers/include/wave/containers/impl/measurement_container_base.hpp

```diff
@@ -0,0 +1,161 @@
+#include <boost/tuple/tuple.hpp>
+
+namespace wave {
+
+/** Construct the container with the contents of a range */
+template <typename Derived>
+template <typename InputIt>
+MeasurementContainerBase<Derived>::MeasurementContainerBase(InputIt first,
+                                                            InputIt last) {
+    this->composite().insert(first, last);
+};
+
+template <typename Derived>
+bool MeasurementContainerBase<Derived>::empty() const noexcept {
+    return this->composite().empty();
+}
+
+template <typename Derived>
+typename MeasurementContainerBase<Derived>::size_type
+MeasurementContainerBase<Derived>::size() const noexcept {
+    return this->composite().size();
+}
+
+template <typename Derived>
+std::pair<typename MeasurementContainerBase<Derived>::iterator, bool>
+MeasurementContainerBase<Derived>::insert(const MeasurementType &m) {
+    return this->composite().insert(m);
+}
+
+template <typename Derived>
+template <typename InputIt>
+void MeasurementContainerBase<Derived>::insert(InputIt first, InputIt last) {
+    return this->composite().insert(first, last);
+}
+
+template <typename Derived>
+template <typename... Args>
+std::pair<typename MeasurementContainerBase<Derived>::iterator, bool>
+MeasurementContainerBase<Derived>::emplace(Args &&... args) {
+// Support Boost.MultiIndex <= 1.54, which does not have emplace()
+#if BOOST_VERSION < 105500
+    return this->composite().insert(
+      MeasurementType{std::forward<Args>(args)...});
+#else
+    return this->composite().emplace(std::forward<Args>(args)...);
+#endif
+}
+
+template <typename Derived>
+template <typename... Args>
+typename MeasurementContainerBase<Derived>::size_type
+MeasurementContainerBase<Derived>::erase(Args &&... args) {
+    auto &composite = this->composite();
```

Comment:
Mainly for readability. I could call `this->composite()` each time, but that's
- longer
- unclear whether the call is the same each time / has side effects